Xamarin.PropertyEditing/OrderedDictionary.cs

```diff
@@ -137,8 +155,17 @@ public IEqualityComparer<TKey> Comparer
 			get { return new KeyValuePair<TKey, TValue> (this.keyOrder[index], this[index]); }
 			set
 			{
-				keyOrder[index] = value.Key;
+				TKey existingKey = this.keyOrder[index];
+				if (!Equals (existingKey, value.Key))
+					this.keyOrder[index] = value.Key;
```

Comment:
This looks wrong. Let's say the dictionary has `{A, B}, {C, D}}` and you do `this[0] = new KVP {C, E}`, wouldn't you get `{{A, B}, {C, E}}` in the dictionary, `[C, C]` in the key order, and notifications that key `A` was replaced with `C` and that `{A, D}` was replaced with `{C, E}`?